src/main/java/uk/gov/hmcts/ccd/domain/service/processor/SearchInputProcessor.java

```diff
@@ -0,0 +1,87 @@
+
+    @Autowired
+    public SearchInputProcessor(final DateTimeFormatParser dateTimeFormatParser,
+                                @Qualifier(DefaultGetCriteriaOperation.QUALIFIER) final GetCriteriaOperation getCriteriaOperation) {
+        this.dateTimeFormatParser = dateTimeFormatParser;
+        this.getCriteriaOperation = getCriteriaOperation;
+    }
+
+    public Map<String, String> execute(String view, MetaData metadata, Map<String, String> queryParameters) {
+        List<? extends CriteriaInput> criteriaInputs =
+            getCriteriaOperation.execute(metadata.getCaseTypeId(), null,
+                view == null ? CriteriaType.SEARCH : CriteriaType.valueOf(view));
+
+        Map<String, String> newParams = new HashMap<>();
+        queryParameters.entrySet().stream().forEach(entry -> {
+            Optional<? extends CriteriaInput> input = criteriaInputs.stream()
+                .filter(i -> i.getField().getId().equals(entry.getKey()))
+                .findAny();
+
+            if (input.isPresent() &&
+                DisplayContextParameter
+                    .hasDisplayContextParameterType(input.get().getDisplayContextParameter(), DisplayContextParameterType.DATETIMEENTRY)) {
+                newParams.put(entry.getKey(),
+                    processValue(entry.getKey(), input.get().getDisplayContextParameter(),
+                        entry.getValue(), input.get().getField().getType()));
+            } else {
+                newParams.put(entry.getKey(), entry.getValue());
+            }
+        });
+
+        return newParams;
+    }
+
+    private String processValue(String id, String displayContextParameter, String value, FieldType fieldType) {
+        try {
+            if (fieldType.getType().equals(FieldType.DATE)) {
+                return dateTimeFormatParser.convertDateToIso8601(format(displayContextParameter, fieldType), value);
+            } else if (fieldType.getType().equals(FieldType.DATETIME)) {
+                return dateTimeFormatParser.convertDateTimeToIso8601(format(displayContextParameter, fieldType), value);
+            } else {
+                return value;
+            }
+        } catch (Exception e) {
+            throw new DataProcessingException().withDetails(
+                String.format("Unable to process input %s with value %s. Expected format: %s",
+                    id,
+                    value,
+                    format(displayContextParameter, fieldType))
+            );
+        }
+    }
+
+    private String format(String displayContextParameter, FieldType fieldType) {
+        return DisplayContextParameter
+            .getDisplayContextParameterOfType(displayContextParameter, DisplayContextParameterType.DATETIMEENTRY)
+            .map(DisplayContextParameter::getValue)
+            .orElseGet(() -> fieldType.getType().equals(FieldType.DATE) ?
+                DateTimeFormatParser.DATE_FORMAT.toString() :
+                DateTimeFormatParser.DATE_TIME_FORMAT.toString());
```

Comment:
is this needed? if there's no datetimeentry then the date must already be in iso format right?